lib/rspec/matchers.rb

```diff
@@ -927,6 +928,18 @@ def method_missing(method, *args, &block)
       end
     end
 
+    if RUBY_VERSION.to_f >= 1.9
+      def respond_to_missing?(method, include_private=false)
+        method = method.to_s
+        method =~ BE_OR_HAS_PREDICATE_REGEX || super
+      end
+    else # for 1.8.7
+      def respond_to?(method, include_private=false)
```

Comment:
`respond_to?` must be public but you're in the section of this module under `private`.  OTOH, `respond_to_missing?` should be private (it's a callback rather than something meant to be called from outside the object, like `respond_to?` is), so my suggestion is just to add a `public :respond_to?` line after this method def to flip it back to public.